src/string.cpp

```diff
@@ -226,32 +226,33 @@ std::string StrMakeValid(std::string_view str, StringValidationSettings settings
 /**
  * Checks whether the given string is valid, i.e. contains only
  * valid (printable) characters and is properly terminated.
- * @param str  The string to validate.
- * @param last The last character of the string, i.e. the string
- *             must be terminated here or earlier.
+ * @note std::span is used instead of std::string_view as this is only used for validating fixed-length string buffers.
```

Comment:
What's maybe even more important why `std::string_view` shouldn't be used because its constructor is going to essentially call `strlen`, which assumes there is a `'\0'`. Exactly one of those things that this function checks to ensure the validity of the passed C-style string.